### ISSUE TITLE
Use ReplaceAll

### ISSUE THREAD

**Issue by breml** [go-critic/go-critic#872]
Since Go version 1.12, `strings.ReplaceAll` and `bytes.ReplaceAll` are available. These should be preferred over their predecessors `Replace` (with `n` set to `-1`), because they make the code more readable.

Before:

```Go
x := strings.Replace(x, "foo", "bar", -1)
y := bytes.Replace(y, []byte("foo"), []byte("bar"), -1)
```

After:

```Go
x := strings.ReplaceAll(x, "foo", "bar")
y := bytes.ReplaceAll(y, []byte("foo"), []byte("bar"))
```

**Comment by cristaloleg** [comment]
Looks like a subcase of `wrapperFunc` checker https://github.com/go-critic/go-critic/blob/a41a1de8e811d54f117bd1ed2916ed31d730d82b/checkers/testdata/wrapperFunc/positive_tests.go#L26

**Comment by quasilyte** [comment]
`wrapperFunc` does that, but unfortunately, it suggests replace all forms even for Go 1.12-.
This is because we have no "target Go version" options right now.

**Comment by breml** [comment]
If I get this correctly, this is already included in go-critic, the only thing that could be improved is, the suggestion should only happen, if a version of Go >= 1.12 is used.
But to add version depended suggestions would need a major change, wouldn't it?

**Comment by quasilyte** [comment]
> But to add version depended suggestions would need a major change, wouldn't it?

I think it's not that hard to add.
But it should be a separate issue probably. As for ReplaceAll, since most users have now go1.12+, the priority is lower than it was before.

**Comment by tehsphinx** [comment]
It would be really nice if the go version in `go.mod` would be used (if present).
